HumanCellAtlas/secondary-analysis
# Replace await_workflow_completion script

### Purpose
This PR replaces some code that duplicates logic in cromwell-tools to poll for workflow completion. That script (` await_workflow_completion.py`) was also using one function from an older version of cromwell-tools.

Fixes: 
- https://broadinstitute.atlassian.net/browse/GH-64
- https://broadinstitute.atlassian.net/browse/GH-101

---
### Changes
Replace script that polls cromwell status. Instead use the cromwell-tools wait cli command.

---
### Review Instructions
This change can be tested locally:

1. Clone https://github.com/HumanCellAtlas/secondary-analysis
2. Check out this branch (`se-fix-cromwell-tools-use`)
3. In `secondary-analysis/tests/integration_test/run_int_test_locally.sh`, change the SECONDARY_ANALYSIS_VERSION to the commit `59cbf76431c29dde3f8f1990e256d1dde8455f46` to use the last working version of the secondary-analysis deploy repo.
4. Run `bash run_int_test_locally.sh <path_to_cloned_secondary_analysis_repo>`
5. Confirm that the test outputs a `SS2_WORKFLOW_ID` and that it is `Awaiting workflow completion` 

---
### PR Checklist
_Please ensure the following when opening a PR:_

- [x] This PR added or updated tests.
- [ ] This PR updated docstrings or documentation.
- [ ] This PR applied Python style guidelines, specifically followed [Google style docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html#example-google) for this repo.
- [ ] This PR considered generalizability beyond our own use case.

---
### Follow-up Discussions
The test will run each workflow sequentially, which means that it will take much longer for the test to complete. Currently, only the SmartSeq2 test is getting run but when multiple pipelines are run in this test, the polling logic should be moved into a separate function that passes in all of the workflow ids.

Additionally, the `wait` function in cromwell-tools does not output any information until all workflows succeed (or if one of them fails) so the test will now say "Awaiting workflow completion" until something happens. Maybe cromwell-tools could log the workflow statuses as it polls cromwell so that it is more informative?
